pkg/kv/kvclient/kvcoord/dist_sender.go

```diff
@@ -1220,6 +1220,19 @@ func (ds *DistSender) divideAndSendBatchToRanges(
 		canParallelize = canParallelize && !isExpensive
 	}
 
+	// Iterate over the ranges that the batch touches. The iteration is done in
+	// key order - the order of requests in the batch is not relevant. Each
```

Comment:
> the order of requests in the batch is not relevant

This is a bit misleading. The order of requests in the batch is still stable relative to other requests that are routed to the same range. Consider clarifying.